scrambles/src/main/java/org/worldcubeassociation/tnoodle/puzzle/SquareOnePuzzle.java

```diff
@@ -196,32 +200,82 @@ protected int getRandomMoveCount() {
         return 40;
     }
 
-    /*
-    // TODO - we can't filter super aggresively until
-    // Chen Shuang's optimal solver is fixed.
     @Override
     protected String solveIn(PuzzleState ps, int n) {
```

Comment:
It took me a while to understand how un-commenting this function could "enable" aggressive filtering, can you please confirm my understanding is correct: when `solveIn` is commented, the generic solving algorithm actually calls the `solveIn` implemented [here](https://github.com/thewca/tnoodle-lib/blob/master/scrambles/src/main/java/org/worldcubeassociation/tnoodle/scrambles/Puzzle.java#L383) which actually implements a BFS valid for all puzzles.

Reply:
Your understanding is correct.